image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
 
-== OpenShift Specific Recommendations
-=== Enable images for Source-To-Image
-For images that are intended to run application code provided by a third party (e.g. a Ruby image designed to run Ruby coded provided by a developer), you should enable your image to work with the https://github.com/openshift/source-to-image[Source to Image] (STI) build tool.  It is a framework which makes it easy to write images that take application source code as an input, and produce a new image that will run the assembled application as output.  For example, this https://github.com/openshift/wildfly-8-centos[Wildfly] image defines STI scripts which will run a `maven` build on a java source repository and copy the resulting war file into the Wildfly deployments directory.  The resulting image will now automatically start Wildfly with the application running.
+[[using-env-vars]]
+*Using Environment Variables for Configuration*
 
-For more details about how to write STI scripts for your image, see the https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md[STI documentation].
+Users of your image should be able to configure it without having to create a downstream image based on your image. This means that the runtime configuration should be handled using environment variables.
+For a simple configuration, the running process can consume the environment variables directly. For a more complicated configuration or for runtimes which do not support this, configure the runtime by defining a template configuration file which is processed during startup. During this processing, values supplied using environment variables can be substituted into the configuration file or used to make decisions about what options to set in the configuration file.
 
-=== Use Services for inter-image communication
-For cases where your image needs to communicate with a service provided by another image (e.g. a web frontend image that needs to access a database image to store/retrieve data), your image should consume an OpenShift Service.  Services provide a static endpoint for access which will not change as containers are stopped/started/moved.  In addition services provide load balancing for requests.  For more information see https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/services.md[this documentation].  (NOTE to docs team:  this link should really go to something in the openshift docs once we have it)
+It is also possible and recommended to pass secrets such as certificates and keys into the container using environment variables. This ensures that the secret values do not end up committed in an image and leaked into a Docker registry.
```

Comment:
[Question]
With the term "secrets" do we mean "sensitive information"? I haven't seen it referred to as "secrets" a lot.

[Suggestion]
It is also possible and we recommend that you pass sensitive information, such as as certificates and keys, into the container using environment variables. This ensures the information does not end up committed to an image and leaked into a Docker registry. 

[Question]
Is it more appropriate to say something is "committed to an image", or something is "committed in an image"?  I think "committed to an image" makes more sense to me, but I may be wrong.